python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
Will this change break the semantics? With this change it only consider running on yarn.